### PR TITLE
[lookup] cleaning constraints

### PR DESCRIPTION
**Pull request by mimoo**
Cleaned up some code as I was going through it.
Low priority, as it will probably mess up a bunch of other PRs from @mrmr1993

### CODE REVIEW

File: kimchi/src/circuits/lookup/constraints.rs [hunk 1]
```diff
@@ -440,71 +440,98 @@ pub fn constraints<F: FftField>(configuration: &LookupConfiguration<F>, d1: D<F>
...
-        .iter()
-        .enumerate()
-        .map(|(i, _)| column(Column::LookupKindIndex(i)))
-        .fold(E::zero(), |acc: E<F>, x| acc + x);
+    // gamma * (beta + 1)
+    let gammabeta1 =
+        E::<F>::Constant(ConstantExpr::Gamma * (ConstantExpr::Beta + ConstantExpr::one()));
+
+    // the nominator part in the multiset check of plookup
+    let nominator = {
```
Comment (mimoo) [inline kimchi/src/circuits/lookup/constraints.rs:448]: moved the code within two main blocks:

```rust
let nominator = { ... };
let denominator = { ... };

let aggreg_equation = E::cell(Column::LookupAggreg, Next) * denominator
    - E::cell(Column::LookupAggreg, Curr) * nominator;
```

File: kimchi/src/circuits/lookup/constraints.rs [hunk 2]
```diff
@@ -440,71 +440,98 @@ pub fn constraints<F: FftField>(configuration: &LookupConfiguration<F>, d1: D<F>
...
+        let t_chunk = gammabeta1.clone()
             + E::cell(Column::LookupTable, Curr)
-            + E::beta() * E::cell(Column::LookupTable, Next));
-
-    let num_rows = d1.size as usize;
+            + E::beta() * E::cell(Column::LookupTable, Next);
 
-    let num_lookup_rows = num_rows - ZK_ROWS - 1;
+        // return the nominator
+        f_chunk * t_chunk
```
Comment (mimoo) [inline kimchi/src/circuits/lookup/constraints.rs:533]: more clearly showing the combination of the f part with the t part

File: kimchi/src/circuits/lookup/constraints.rs [hunk 3]
```diff
@@ -440,71 +440,98 @@ pub fn constraints<F: FftField>(configuration: &LookupConfiguration<F>, d1: D<F>
...
-        .kinds
-        .iter()
-        .enumerate()
-        .map(|(i, _)| column(Column::LookupKindIndex(i)))
-        .fold(E::zero(), |acc: E<F>, x| acc + x);
+    // gamma * (beta + 1)
+    let gammabeta1 =
+        E::<F>::Constant(ConstantExpr::Gamma * (ConstantExpr::Beta + ConstantExpr::one()));
+
+    // the nominator part in the multiset check of plookup
```
[the remark this answers is not in the document]
Comment (mimoo) [reply kimchi/src/circuits/lookup/constraints.rs:447]: oh dang, yeah I never spell this word correctly ;_;

File: kimchi/src/circuits/constraints.rs [hunk 4]
```diff
@@ -344,14 +344,14 @@ impl<F: FftField + SquareRootField> LookupConstraintSystem<F> {
                 let lookup_table = lookup_tables.into_iter().next().unwrap();
 
                 // get the last entry in each column of each table
-                let dummy_lookup_value: Vec<F> =
+                let dummy_lookup_row: Vec<F> =
```
[the remark this answers is not in the document]
Comment (mimoo) [reply kimchi/src/circuits/constraints.rs:347]: ah yeah, not a bad idea, I'll change to `dummy_lookup_entries`, my understanding was that here it's not joined yet, but I'll re-read the code

**Review by mrmr1993 — APPROVED** [review]
1 nit, but otherwise lgtm

File: kimchi/src/circuits/lookup/constraints.rs [hunk 5]
```diff
@@ -532,20 +559,44 @@ pub fn constraints<F: FftField>(configuration: &LookupConfiguration<F>, d1: D<F>
     // and if i % 2 = 1, we enforce that the
     // first element of LookupSorted(i) = first element of LookupSorted(i + 1)
 
-    let s_chunk = (0..(lookup_info.max_per_row + 1))
+    let sorted_size = lookup_info.max_per_row + 1 /* for the XOR lookup table */;
```
Comment (mrmr1993) [inline kimchi/src/circuits/lookup/constraints.rs:562]: I would remove `XOR` here, if we're intending to go with a combined lookup table, a la #381.

Comment (mimoo) [reply kimchi/src/circuits/lookup/constraints.rs:562]: to my previous point, that code would definitely have to change if we start supporting multiple lookup tables, so in the mean time let's make it explicit that it's only working for one table, and for the XOR table :o

**Pull request merged**